tests/build_rpms.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+# Copied from https://github.com/htcondor/htcondor-ce
+
+set -exu
+
+# Build source and binary RPMs.
+# SRPMs will be /tmp/rpmbuild/SRPMS/*.rpm.
+# Binary RPMs will be /tmp/rpmbuild/RPMS/*/*.rpm.
+
+OS_VERSION=$1
+BUILD_ENV=$2
+
+# Create the condor user/group for subsequent chowns,
+# using the official RHEL UID/GID
+groupadd -g 64 -r condor
+useradd -r -g condor -d /var/lib/condor -s /sbin/nologin \
+        -u 64 -c "Owner of HTCondor Daemons" condor
+
+if  [[ $OS_VERSION == 7 ]]; then
+    YUM_PKG_NAME="yum-plugin-priorities"
+else
+    YUM_PKG_NAME="yum-utils"
+fi
+
+# Clean the yum cache
+yum clean all
+
+# Exponential backoff retry for missing DNF lock file
+# FileNotFoundError: [Errno 2] No such file or directory: '/var/cache/dnf/metadata_lock.pid'
+# 2020-03-17T13:43:57Z CRITICAL [Errno 2] No such file or directory: '/var/cache/dnf/metadata_lock.pid'
+set +e
+for retry in {1..3}; do
+    yum -y -d0 update  # Update the OS packages
+    [[ $? -eq 0 ]] && break
+    sleep $((5**$retry))
+done
+set -e
```

Comment:
Fine how you have it - but note that `set -e` does not trigger an exit on failure when the failed command is on the left side of a `&&` or `||`; so you can equivalently do something like

```suggestion
for retry in {1..3}; do
    yum -y -d0 update && break  # Update the OS packages
    sleep $((5**$retry))
done
```